Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java

```diff
@@ -102,63 +89,62 @@ public CONNECTSamlAssertionValidator(PropertyAccessor propertyAccessor) {
-            ValidatorSuite schemaValidators = org.opensaml.Configuration.getValidatorSuite("saml1-schema-validator");
-            ValidatorSuite specValidators = org.opensaml.Configuration.getValidatorSuite("saml1-spec-validator");
-            try {
-                schemaValidators.validate(assertion.getSaml1());
-                specValidators.validate(assertion.getSaml1());
-            } catch (ValidationException e) {
-                LOG.debug("Saml Validation error: " + e.getMessage(), e);
-                throw new WSSecurityException(WSSecurityException.FAILURE, "invalidSAMLsecurity");
-            }
+            assertion.validateSignatureAgainstProfile();
         } else if (assertion.getSaml2() != null) {
-            List<ValidatorSuite> validators = new LinkedList<>();
-            validators.add(org.opensaml.Configuration.getValidatorSuite("saml2-core-schema-validator"));
-            validators.addAll(getSaml2SpecValidators());
-
-            for (AuthzDecisionStatement auth : assertion.getSaml2().getAuthzDecisionStatements()) {
-                if (StringUtils.isBlank(auth.getResource())) {
-                    auth.setResource(TEMP_RESOURCE_FOR_VALIDATION);
-                }
-            }
-
-            Issuer issuer = assertion.getSaml2().getIssuer();
-            if (issuer.getFormat().equals("urn:oasis:names:tc:SAML:1.1:nameid-format:entity")) {
+            /*
+             * for (final AuthzDecisionStatement auth : assertion.getSaml2().getAuthzDecisionStatements()) {
+             *
+             * Please see section 3.2.3.1 on why it should be empty or in uri format if
+             * (StringUtils.isBlank(auth.getResource())) {
+             * LOG.debug("since this is empty in resource for AuthzDecisionStatement, we will set temporary value");
+             * auth.setResource(TEMP_RESOURCE_FOR_VALIDATION); }
+             *
+             * if(StringUtils.isBlank(auth.getResource())) {
+             * LOG.debug("since this is empty in resource for AuthzDecisionStatement, we will set temporary value");
+             * auth.setResource("http://localhost:8080/test"); assertion.getSaml2().set } }
+             */
+
+            final Issuer issuer = assertion.getSaml2().getIssuer();
+            if ("urn:oasis:names:tc:SAML:1.1:nameid-format:entity".equals(issuer.getFormat())) {
                 if (!StringUtils.isBlank(issuer.getSPProvidedID())) {
-                    throw new WSSecurityException("SOAP header element Security/Assertion/Issuer/@Format = " + ""
-                            + "urn:oasis:names:tc:SAML:1.1:nameid-format:entity" + "" + "and"
-                            + "Security/Assertion/Issuer/@SPProvidedID" + " " + "is present.");
+                    throw new WSSecurityException(ErrorCode.FAILED_CHECK,
+                            "SOAP header element Security/Assertion/Issuer/@Format = " + ""
+                                    + "urn:oasis:names:tc:SAML:1.1:nameid-format:entity" + "" + "and"
```

Comment:
Here (and in the lines below), consider if constants may be more appropriate for some of these strings (e.g., the namespace, or the common XPath nodes).